packages/ra-core/src/reducer/admin/resource/list/ids.ts

```diff
@@ -28,12 +28,18 @@ type State = IdentifierArrayWithDate;
 
 export const addRecordIdsFactory = getFetchedAtCallback => (
     newRecordIds: IdentifierArrayWithDate = [],
-    oldRecordIds: IdentifierArrayWithDate
+    oldRecordIds: IdentifierArrayWithDate,
+    preserveOldIds: boolean = false
 ): IdentifierArrayWithDate => {
-    const newFetchedAt = getFetchedAtCallback(
+    let newFetchedAt = getFetchedAtCallback(
```

Comment:
Is there a reason to replace `const` by `let` ?

Reply:
@fargito yes, reassigning value to that variable.

Reply:
Wouldn't it be cleaner to create another variable named for example `returnedFetchedAt` that would contain, depending on `preserveOldIds`, either `{ ...oldRecordIds.fetchedAt, ...newFetchedAt }` or `{ ...newFetchedAt }` ? And then pass it as argument to the `filter` function ?

Reply:
Yes, makes sense, will change and update it.